nemo/collections/asr/metrics/multitask.py

```diff
@@ -0,0 +1,237 @@
+            cuts_idx = None
+            if cuts is not None:
+                indices_cpu = indices.cpu().tolist()
+                cuts_idx = [cuts[idx] for idx in indices_cpu]
+
+            metric.update(
+                predictions=predictions[indices],
+                predictions_lengths=predictions_lengths[indices],
+                predictions_mask=predictions_mask[indices],
+                targets=targets[indices],
+                targets_lengths=targets_lengths[indices],
+                input_ids=input_ids[indices],
+                cuts=cuts_idx,
+            )
+
+    def compute(self, return_all_metrics=False, prefix="", suffix=""):
+        output_dict = {}
+        for name, metric in self._metric_dict.items():
+            # Check if update marked this metric empty for batch
+            # Since ASR models do full loop per step this ignores breaking behavior
+            if self._skip_dict[name]:
+                self._skip_dict[name] = False
+                continue
+
+            # TODO: Change behavior of WER
+            # so it has a dict output
+            if name == "wer":
+                wer, wer_num, wer_denom = metric.compute()
+                if return_all_metrics:
+                    output_dict.update(
+                        {
+                            f"{prefix}wer{suffix}": wer,
+                            f"{prefix}wer_num{suffix}": wer_num,
+                            f"{prefix}wer_denom{suffix}": wer_denom,
+                        }
+                    )
+                else:
+                    output_dict.update(
+                        {
+                            f"{prefix}wer{suffix}": wer,
+                        }
+                    )
+            else:
+                output_dict.update(
+                    metric.compute(
+                        return_all_metrics=return_all_metrics,
+                        prefix=prefix,
+                        suffix=suffix,
+                    )
+                )
+        return output_dict
+
+    def reset(self):
+        {metric.reset() for name, metric in self._metric_dict.items()}
+
+    # TODO: Add properties to `Canary` to simply return `task` idx.
+    def _canary_index_fn(self, prompt_ids: torch.Tensor, slots: Dict[str, (str | bool)]) -> torch.Tensor:
+        if slots["task"] in TASK_TRANSLATE:
+            # 1 -> `source_lang` in canary, 3 -> 'target_lang. Use these instead of task ID to avoid lookup.
+            condition_met = prompt_ids[:, 1] != prompt_ids[:, 3]
+        else:  # default to transcribe
+            condition_met = prompt_ids[:, 1] == prompt_ids[:, 3]
+        indices = torch.nonzero(condition_met, as_tuple=False)
+        # reshape in case 0 dim
+        return indices.reshape(indices.numel())
+
+    # TODO: Add properties to `Canary2` to simply return `task` idx.
+    def _canary2_index_fn(self, prompt_ids: torch.Tensor, slots: Dict[str, str | bool]) -> torch.Tensor:
```

Comment:
This definitely needs good test cases for all possible cases

Reply:
What ya feeling unit test wise? Just assert indexes line up with a dummy lhotse dataloader?